src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -437,33 +454,131 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
             }
         }
 
-        public static unsafe void ScaleU(float scale, Span<float> dst)
+        public static unsafe void Scale(float scale, Span<float> dst)
         {
-            fixed (float* pdst = dst)
+            fixed (uint* pLeadingAlignmentMask = &LeadingAlignmentMask[0])
+            fixed (uint* pTrailingAlignmentMask = &TrailingAlignmentMask[0])
+            fixed (float* pd = dst)
             {
-                float* pDstCurrent = pdst;
-                float* pEnd = pdst + dst.Length;
-
-                Vector128<float> scaleVector = Sse.SetAllVector128(scale);
+                float* pDstCurrent = pd;
+                int length = dst.Length;
+                Vector128<float> scaleVector128 = Sse.SetAllVector128(scale);
 
-                while (pDstCurrent + 4 <= pEnd)
+                if (length < 4)
                 {
-                    Vector128<float> dstVector = Sse.LoadVector128(pDstCurrent);
+                    // Handle cases where we have less than 128-bits total and can't ever use SIMD acceleration.
+                    switch (length)
+                    {
+                        case 3: dst[2] *= scale; goto case 2;
+                        case 2: dst[1] *= scale; goto case 1;
+                        case 1: dst[0] *= scale; break;
+                    }
+                    return;
+                }
 
-                    dstVector = Sse.Multiply(scaleVector, dstVector);
-                    Sse.Store(pDstCurrent, dstVector);
+                nuint address = (nuint)(pd);
+                int misalignment = (int)(address % 16);
+                int remainder = 0;
 
-                    pDstCurrent += 4;
-                }
+                if ((misalignment & 3) != 0)
+                {
+                    // Handles cases where the data is not 32-bit aligned and we can't ever use aligned operations
+                    remainder = length % 4;
```

Comment:
I don't think this `remainder` is handled anywhere. It looks like it is handled inside the `else` below, which won't get hit in this case.

Reply:
Good catch, the block that handles the remainder is in the wrong place.